python-package/lightgbm/basic.py

```diff
@@ -1375,6 +1378,22 @@ def get_init_score(self):
             self.init_score = self.get_field('init_score')
         return self.init_score
 
+    def get_data(self):
+        """Get the raw data of the Dataset.
+
+        Returns
+        -------
+        data : string, numpy array, pandas DataFrame, scipy.sparse, list of numpy arrays or None
```

Comment:
What about `string` and `list of numpy arrays`? Hot to "slice" them?..

Reply:
I think we cannot slide them, maybe we should throw a warning or error here.